core/whistle_couch-1.0.0/src/couch_util.erl

```diff
@@ -728,13 +734,21 @@ do_save_docs(#db{}=Db, Docs, Options, Acc) ->
 perform_save_docs(Db, Docs, Options) ->
     PreparedDocs = [maybe_set_docid(D) || D <- Docs],
     _ = flush_cache_docs(Db, PreparedDocs),
-    case ?RETRY_504(couchbeam:save_docs(Db, PreparedDocs, Options)) of
+    case ?RETRY_504(couchbeam:save_docs(Db, maybe_tombstones(PreparedDocs), Options)) of
         {'ok', JObjs} ->
             _ = maybe_publish_docs(Db, PreparedDocs, JObjs),
             {'ok', JObjs};
         {'error', _}=E -> E
     end.
 
+-spec maybe_tombstones(wh_json:objects()) -> wh_json:objects().
+maybe_tombstones(Docs) ->
+    [ maybe_tombstone(wh_json:get_value(<<"_deleted">>, Doc), Doc) || Doc  <- Docs ].
```

Comment:
I'd use `wh_json:is_true(<<"_deleted">>, Doc)` and match 'true' and 'false' in `maybe_tombstone/2`. I'm a bit worried that this will operate on all requests.